features/fastreflection/proto_message.go

```diff
@@ -280,7 +289,51 @@ func (g *fastGenerator) genProtoMethods() {
 	g.P(`// "google.golang.org/protobuf/runtime/protoiface".Methods.`)
 	g.P("// Consult the protoiface package documentation for details.")
 	g.P("func (x *", g.typeName, ") ProtoMethods() *", protoifacePkg.Ident("Methods"), " {")
-	g.P("return nil")
+
+	// SIZE METHOD
+	g.P(`size := func(input `, protoifacePkg.Ident("SizeInput"), ") ", protoifacePkg.Ident("SizeOutput"), " {")
+	g.P("s := input.Message.(*", fastReflectionTypeName(g.message), ").Unwrap().Size()")
+	g.P("return ", protoifacePkg.Ident("SizeOutput"), "{")
+	g.P("NoUnkeyedLiterals: struct{}{},")
+	g.P("Size: s,")
+	g.P("}")
+	g.P("}")
+
+	// MARSHAL METHOD
+	g.P(`marshal := func(input `, protoifacePkg.Ident("MarshalInput"), `) (`, protoifacePkg.Ident("MarshalOutput"), `, error) {`)
+	g.P(`bz, err := input.Message.(*`, fastReflectionTypeName(g.message), `).Unwrap().Marshal()`)
```

Comment:
I'm not sure if the Marshal/Unmarshal methods should be API exposed by the message itself.

An alternative could be:

```go
var _AMethods = protoiface.Methods{
Marshal: _AMethodMarshal
... etc.
}

func _AMethodMarshal(MarshalInput) (MarshalOutput, error) { logic ... }

func (a *fastReflection_A) ProtoMethods() *protoiface.Methods { return _AMethods }
```

IDK if @aaronc has any opinion on this.

Reply:
i do see some advantage for having marshal code exported, for example if someone simply has the message and wants to call marshal directly on it, rather than deal with calling ProtoMethods() and then dealing with MarshalInput etc